### PR TITLE
Use GITHUB_TOKEN, default branch based and simplified periodically check process

### LINKED ISSUE #496: [CI] Use default Actions token for wf triggering
### What is the expected enhancement?
Use default Actions token for triggering actions and not using anymore personal token.

### PR DESCRIPTION
**Pull request by mickahell**
### Summary

Replace custom token with `GITHUB_TOKEN` and `main` branch with default checkout branch.
Simplified the periodically check process.

### Details and comments

- [x] Update `GITHUB_TOKEN`
- [x] Delete ref to so call `main` branch
- [x] Add `process_temp_test_results_files` into wf `ecosystem-project-check.yml`
- [x] Make every test kind into same action
- [x] Tests --> https://github.com/mickahell/ecosystem/pull/186

---
Closes #496

### CODE REVIEW

**Comment by mickahell** [comment]
@frankharkins ready for your review :)

It has a bit more than just the token, I simplified the process and clean some CI

File: .github/workflows/ecosystem-batch-repo-check.yml [hunk 1]
```diff
@@ -50,6 +49,6 @@ jobs:
                             --repo_url=${repositories[$i]} \
                             --branch_name=${{ steps.vars.outputs.pr_branch_name }} \
                             --tier=${tiers[$i]} \
-                            --token=${{ secrets.WORKFLOW_DISPATCH_PAT }} \
+                            --token=${{ secrets.GITHUB_TOKEN }} \
```
Comment (frankharkins) [inline .github/workflows/ecosystem-batch-repo-check.yml:52]: From "[Using the GITHUB_TOKEN in a workflow](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow)":

> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN, with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run.

This means this action can't be triggered as `schedule`? I'm not sure this is a suitable workaround; or are you thinking we just trigger manually until we work out what the token problem is?

Comment (mickahell) [reply .github/workflows/ecosystem-batch-repo-check.yml:52]: Yeah we are fine the wf triggered is a `repository_dispatch` :)
https://github.com/qiskit-community/ecosystem/blob/9ff2fb7c54e2515010860e34f182bc64bb581a23/.github/workflows/ecosystem-project-check.yml#L12-L13

This means the GITHUB_TOKEN can't trigger anything else than `workflow_dispatch` and `repository_dispatch` so in our case we are good :)

Comment (mickahell) [reply .github/workflows/ecosystem-batch-repo-check.yml:52]: I let click `Resolve conversion` when you see my response ;)

Comment (frankharkins) [reply .github/workflows/ecosystem-batch-repo-check.yml:52]: Ok I see, thanks

**Review by frankharkins — APPROVED** [review]
Thanks! The action looks neater too 😄

**Pull request merged**